ably/realtime_channel.go

```diff
@@ -724,9 +774,20 @@ func (c *RealtimeChannel) ErrorReason() *ErrorInfo {
 }
 
 func (c *RealtimeChannel) notify(msg *protocolMessage) {
+	// RTL15b
+	if !empty(msg.ChannelSerial) && msg.Action == actionMessage ||
+		msg.Action == actionPresence || msg.Action == actionAttached {
```

Comment:
I suspect you didn't intend the non-empty check to apply only to MESSAGE protocolmessages (&& binds tighter than ||). Use parens

Reply:
updated